### ISSUE TITLE
[Website] Proposal to add Prettier to the project.

### ISSUE THREAD

**Issue by amin-xiv** [multiverseweb/Dataverse#321]
Have an idea for a new feature? Please fill out the sections below. 😊

### Current Behaviour

Currently, there is no 'standard' in terms of formatting code.

### Proposed Solution

Adding a prettier setup to the project, since it will standardize lots of the formatting on save, which will make the process of collaboration easier and smoother.
If I get assigned to do this (hopefully), I would like to be provided with these values:
-  ```printWidth``` : The line length where Prettier will try to wrap the code. If the line is longer than this value, Prettier will attempt to break it into multiple line.
- ```tabWidth``` :  Determines how many spaces Prettier uses for indentation. (Preferably 4).
- ```useTabs``` : Can be ```true```/```false```, helps prettier determine whether to use spacing or tabs for indentation.
- ```bracketSpacing```` : Can either be ```true```/```false``, if it is true, braces/brackets will seem like this  ```{ value: key }```, if false it will seem like this: ```{value: key}```.
- ```arrowPens``` : can either be ```true```/```false```, determines whether to wrap perimeters of arrow functions with parenthesis or no even though if it has exactly one parameter.
- ```trailingComma``` : can either be ```true```/```false```, simply determines whether to place trailing commas or no.

Those are simply the ones I thought that might relevant and useful. I would like to hear your opinion.

**Comment by amin-xiv** [comment]
@multiverseweb Any thoughts about this? 😊

**Comment by multiverseweb** [comment]
Okay @amin-xiv, make sure it doesn't affect the functionality of the code. Also will it run for all files or only for python?

**Comment by multiverseweb** [comment]
> Here are the values:
> printWidth : (doubt: it should be the number of characters per line right?)
> tabWidth : 4
> useTabs : true
> bracketSpacing : false
> arrowPens : true
> trailingComma : false

@amin-xiv I accidentally edited your comment instead of quoting a reply 🥲

**Comment by amin-xiv** [comment]
@multiverseweb 1/No it doesn't affect the functionality. 2/Prettier doesn't work for python, it works for JS, TS, CSS, HTML, markdown, YAML and JSON. It simply arranges the code, according to your configuration, you can define the variables that was stated above. Thank you.

**Comment by amin-xiv** [comment]
@multiverseweb And yes printWidth, are the characters per line that require the extension to automatically wrap the rest of the text to the next line.

**Comment by multiverseweb** [comment]
> @multiverseweb And yes printWidth, are the characters per line that require the extension to automatically wrap the rest of the text to the next line.

Okay @amin-xiv go ahead. Set the appropriate value of  printWidth yourself, assigned.

**Comment by amin-xiv** [comment]
@multiverseweb Thank you very much sir.

**Comment by amin-xiv** [comment]
@multiverseweb Sorry, but what about ``singleQuote``` , if it is set to ``true``, it will change every double quote to singe, and when it is ``false``, it will do the opposite. 
Note: this setting doesn't affect template strings or HTML attributes.

**Comment by amin-xiv** [comment]
@multiverseweb So far I have added it to the project, what is only remaining is to update the README.md file.

**Comment by multiverseweb** [comment]
@amin-xiv nice, I would like to know what changes are required in the README.md file?

**Comment by amin-xiv** [comment]
@multiverseweb In order to allow prettier:
1/The prettier extension must be installed, so that will be included.
2/In order for it to work, correctly, with the correct version and settings, the ``node_modules/`` folder has to be installed, so I will just add after you clone the repository in the README file, simply run ``npm install`` which is going to install all the dependencies required for it to work across al users. I added the ``node_modules/`` to the ``.gitignore`` folder because it can add unnecessary space, and the user only needs the ``package.json`` and the ``package-lock.json`` folders in order to install it, so they will be included.

In conclusion, what will be added to the README file are going to be some few changes in the section of ``Prerequisites`` in order to set it up correctly.

**Comment by multiverseweb** [comment]
Okay @amin-xiv. In a few days from now I'll be fixing the structure of the repo and automating the process of project set-up. For this, I'll add a file requirements.txt which will include all the modules that need to be installed before using the software.

**Comment by amin-xiv** [comment]
@multiverseweb Okay, Thanks.

**Comment by Hacker-smkg** [comment]
hey is this issue still open ? @multiverseweb

**Comment by multiverseweb** [comment]
Hey @Hacker-smkg, yes this issue is still open. Would you like to work on it?

**Comment by Hacker-smkg** [comment]
What is the requirements to enhance the project can you tell me elaborately

**Comment by multiverseweb** [comment]
Hi @Hacker-smkg, would you like to know the requirements of this `prettier` issue or of the whole project in general?

**Comment by Hacker-smkg** [comment]
yeah i want to know the requirements of this prettier issue @multiverseweb

**Comment by multiverseweb** [comment]
Hey @Hacker-smkg, we'll have to postpone solving this issue, as there are several PRs to be merged that will change the repo structure. We will come back to this issue once the dust is settled.
